### PR TITLE
feat: add `banner` to `ClientUser.edit`

### PR DESCRIPTION
**Pull request by Braandn**
Added "banner" to the edit function.

Till now the official discord docs are not updated, but its is working.

ref: https://github.com/discord/discord-api-docs/pull/6710

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

File: discord/user.py [hunk 1]
```diff
@@ -422,20 +422,24 @@ def _update(self, data: UserPayload) -> None:
...
         .. note::
 
-            To upload an avatar, a :term:`py:bytes-like object` must be passed in that
+            To upload an avatar or banner, a :term:`py:bytes-like object` must be passed in that
             represents the image being uploaded. If this is done through a file
             then the file must be opened via ``open('some_filename', 'rb')`` and
             the :term:`py:bytes-like object` is given through the use of ``fp.read()``.
 
-            The only image formats supported for uploading is JPEG and PNG.
+            The only image formats supported for uploading is JPEG and PNG and GIF.
```
Comment (plun1331) [inline discord/user.py:442]:
```suggestion
            The only image formats supported for uploading are JPEG, PNG, and GIF.
```

File: discord/user.py [hunk 2]
```diff
@@ -422,20 +422,24 @@ def _update(self, data: UserPayload) -> None:
...
 
         .. versionchanged:: 2.0
             The edit is no longer in-place, instead the newly edited client user is returned.
```
Comment (plun1331) [inline discord/user.py:445]: Add a version changed
```suggestion
        .. versionchanged:: 2.0
            The edit is no longer in-place, instead the newly edited client user is returned.
            
        .. versionchanges:: 2.6
            The ``banner`` keyword-only parameter was added.
```

**Review by plun1331 — CHANGES_REQUESTED** [review]
Please add a changelog

File: discord/user.py [hunk 3]
```diff
@@ -422,31 +422,41 @@ def _update(self, data: UserPayload) -> None:
...
             then the file must be opened via ``open('some_filename', 'rb')`` and
             the :term:`py:bytes-like object` is given through the use of ``fp.read()``.
 
-            The only image formats supported for uploading is JPEG and PNG.
+            The only image formats supported for uploading are JPEG, PNG, and GIF.
 
         .. versionchanged:: 2.0
             The edit is no longer in-place, instead the newly edited client user is returned.
 
+        .. versionchanges:: 2.6
```
Comment (plun1331) [inline discord/user.py:447]: mb
```suggestion
        .. versionchanged:: 2.6
```

**Review by Lulalaby — CHANGES_REQUESTED** [review]
Changelog

**Review by plun1331 — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**